src/Akeneo/Pim/Automation/DataQualityInsights/back/Infrastructure/Persistence/Repository/ProductScoreRepository.php

```diff
@@ -51,28 +53,42 @@ public function saveAll(array $productsScores): void
             return;
         }
 
-        $queries = '';
-        $queriesParameters = [];
-        $queriesParametersTypes = [];
+        $insertValues = implode(', ', array_map(function (Write\ProductScores $productScore) {
+            return sprintf(
+                "(%d, '%s', '%s')",
+                $productScore->getProductId()->toInt(),
+                $productScore->getEvaluatedAt()->format('Y-m-d'),
+                \json_encode($productScore->getScores()->toNormalizedRates())
+            );
+        }, $productsScores));
 
-        foreach ($productsScores as $index => $productScore) {
-            Assert::isInstanceOf($productScore, Write\ProductScores::class);
-            $productId = sprintf('productId_%d', $index);
-            $evaluatedAt = sprintf('evaluatedAt_%d', $index);
-            $scores = sprintf('scores_%d', $index);
+        $this->dbConnection->executeQuery(
+            <<<SQL
+INSERT INTO pim_data_quality_insights_product_score (product_id, evaluated_at, scores) 
+VALUES $insertValues AS product_score_values
+ON DUPLICATE KEY UPDATE evaluated_at = product_score_values.evaluated_at, scores = product_score_values.scores;
+SQL
+        );
 
-            $queries .= <<<SQL
-INSERT INTO pim_data_quality_insights_product_score (product_id, evaluated_at, scores)
-VALUES (:$productId, :$evaluatedAt, :$scores)
-ON DUPLICATE KEY UPDATE evaluated_at = :$evaluatedAt, scores = :$scores;
-SQL;
-            $queriesParameters[$productId] = $productScore->getProductId()->toInt();
-            $queriesParametersTypes[$productId] = \PDO::PARAM_INT;
-            $queriesParameters[$evaluatedAt] = $productScore->getEvaluatedAt()->format('Y-m-d');
-            $queriesParameters[$scores] = \json_encode($productScore->getScores()->toNormalizedRates());
-        }
+        // We need to delete younger product scores after inserting the new ones,
+        // so we insure to have 1 product score per product
+        $deleteValues = implode(', ', array_map(function (Write\ProductScores $productScore) {
+            return sprintf(
+                "%d",
+                $productScore->getProductId()->toInt(),
+            );
+        }, $productsScores));
```

Comment:
Do we try to cast the product id in a string? Could we simplify by the following code?
```suggestion
        $deleteValues = implode(', ', array_map(fn (Write\ProductScores $productScore) => (string) $productScore->getProductId()->toInt(), $productsScores));
```